src/Cardano/Wallet/Primitive/Signing.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module provides functionality for signing transactions.
+--
+-- It relies on the binary CBOR format of transactions, an AddressScheme
+-- for deriving address private keys, and cardano-crypto for the actual signing.
+module Cardano.Wallet.Primitive.Signing
+    ( -- * Sign transactions
+      mkStdTx
+    , SignTxError (..)
+    )
+    where
+
+
+import Prelude
+
+import Cardano.Wallet.Binary
+    ( TxWitness (..), encodeTx, toByteString )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (AddressK, RootK)
+    , Key
+    , Passphrase (..)
+    , XPrv
+    , XPub
+    , getKey
+    , publicKey
+    )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( AddressScheme (keyFrom) )
+import Cardano.Wallet.Primitive.Types
+    ( Address, Hash (..), Tx (..), TxIn, TxOut (..) )
+import Control.Monad
+    ( forM )
+import Crypto.Hash
+    ( Blake2b_256, hash )
+import Data.ByteArray
+    ( convert )
+import Data.ByteString
+    ( ByteString )
+
+import qualified Cardano.Crypto.Wallet as CC
+import qualified Codec.CBOR.Encoding as CBOR
+
+
+newtype SignTxError = KeyNotFoundForAddress Address
+
+-- | Construct a standard transaction
+--
+-- " Standard " here refers to the fact that we do not deal with redemption,
+-- multisignature transactions, etc.
+--
+mkStdTx
+    :: AddressScheme s
+    => s
+    -> (Key 'RootK XPrv, Passphrase "encryption")
+    -> [(TxIn, TxOut)]
+    -- ^ Selected inputs
+    -> [TxOut]
+    -- ^ Selected outputs (including change)
```

Comment:
We are missing the change outputs here. Those are part of the `CoinSelection` but are simply `Coins`. In practice, we do need to generate change addresses on the internal chain for all change, in sequence.

However, we also want to leave that flexible in the long-run in order to let users pick their own change addresses if needed.

Reply:
Ah, assumed they were included in the `CoinSelection` outputs, but makes a lot of sense

> However, we also want to leave that flexible in the long-run in order to let users pick their own change addresses if needed.
Sounds good to me

Adding a function
```
generateChangeOutput :: Coin -> TxOut
```
to `AddressScheme`?

Or have it specific to `SeqState`?

Reply:
It is indeed specific to the scheme state. Your suggestion is sound. We'll see later how we enable this to be set by users :+1: